packages/jest-mock/src/index.ts

```diff
@@ -794,39 +794,39 @@ export class ModuleMocker {
         }
       };
 
-      f.mockReturnValueOnce = (value: ReturnType<T>) =>
+      f.mockReturnValueOnce = (value?: ReturnType<T>) =>
         // next function call will return this value or default return value
         f.mockImplementationOnce(() => value);
 
-      f.mockResolvedValueOnce = (value: ResolveType<T>) =>
+      f.mockResolvedValueOnce = (value?: ResolveType<T>) =>
         f.mockImplementationOnce(() =>
           this._environmentGlobal.Promise.resolve(value),
         );
 
-      f.mockRejectedValueOnce = (value: unknown) =>
+      f.mockRejectedValueOnce = (value?: unknown) =>
         f.mockImplementationOnce(() =>
           this._environmentGlobal.Promise.reject(value),
         );
 
-      f.mockReturnValue = (value: ReturnType<T>) =>
+      f.mockReturnValue = (value?: ReturnType<T>) =>
         // next function call will return specified return value or this one
         f.mockImplementation(() => value);
 
-      f.mockResolvedValue = (value: ResolveType<T>) =>
+      f.mockResolvedValue = (value?: ResolveType<T>) =>
         f.mockImplementation(() =>
           this._environmentGlobal.Promise.resolve(value),
         );
 
-      f.mockRejectedValue = (value: unknown) =>
+      f.mockRejectedValue = (value?: unknown) =>
         f.mockImplementation(() =>
           this._environmentGlobal.Promise.reject(value),
         );
 
-      f.mockImplementationOnce = (fn: T) => {
+      f.mockImplementationOnce = (fn?: T) => {
         // next function call will use this mock implementation return value
         // or default mock implementation return value
         const mockConfig = this._ensureMockConfig(f);
-        mockConfig.specificMockImpls.push(fn);
+        fn && mockConfig.specificMockImpls.push(fn);
```

Comment:
I think instead of changing the logic, you should type `specificMockImpls` as `Array<Function | undefined>` here:

https://github.com/facebook/jest/blob/804608713fd3dd6c02bb0a745b9df68a7c91ae76/packages/jest-mock/src/index.ts#L256-L258

Looks like typings are simply incorrect, because just above we have `mockImpl: Function | undefined`. If mock implementation can be `undefined`, specific implementation (`once`) should work in similar fashion. At least that is my feeling. A unit test should show if that is correct idea.